### ISSUE TITLE
Cannot create Kubernetes cluster in a differnet vnet/subnet 

### ISSUE THREAD

**Issue by eynet** [Azure/ACS#84]
Unable to create azure container services (Kubernetes) on a specific VNET using flags '--master-vnet-subnet-id' & ' --agent-vnet-subnet-id' , Look like it's creating it's on VNET other than using the specified ones. 

Trying to spinup new cluster through Azure-CLI uisng following command :
----------
$ az acs create --orchestrator-type kubernetes --resource-group GROUP1--name NAME1 --master-vnet-subnet-id "/subscriptions/<id>/resourceGroups/<resource group name>/providers/Microsoft.Network/virtualNetworks/<vnet name>/subnets/<subnet name>" --generate-ssh-keys
-----------

**Comment by JackQuincy** [comment]
You are missing some of the required parameters for custom vnet. You also need to specify the agent vnet subnet, --agent-vnet-subnet-id, and what address to put the first master ip at, --master-first-consecutive-static-ip.

**Comment by eynet** [comment]
Have tried it , But no luck. Will [Azure VNET plugins ](https://github.com/Azure/azure-container-networking/blob/master/docs/acs.md) helps?

**Comment by JackQuincy** [comment]
That could help but it wouldn't fix this. So it is still provisioning a new vnet and putting the machines in it? What region are you trying to deploy into? Vnet is only supported in our preview regions.  https://github.com/Azure/ACS/blob/master/announcements/2017-08-04_additional_regions.md

**Comment by eynet** [comment]
Yes, It's provisioning new **vnet** , not using the original vnet/subnet given. 

I forget to mention it , ACS is hosted in **westcentralus** which is one of the supported regions.

**Comment by gmartinoreply** [comment]
Hi,
Could you update supported regions? 
I have the same problem as eynet, anyone has resolved?

Thx

**Comment by bhood4** [comment]
Hi,
I think I have the same problem.  I am using terraform with https://github.com/bhood4/terraform-provider-azurerm fork .  I have added vnet_subnet_id on master_profile and agent_profile, and now added master_profile.first_consecutive_static_ip.  I get this error back from the go sdk call:
	
{"code":"DeploymentFailed",
"message":"At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-debug for usage details.",
"details":[
{"code":"BadRequest",
"message":"{\r\n \"error\": {\r\n \"code\": \"InvalidTemplate\",\r\n \"message\": \"Unable to process template language expressions for resource '/subscriptions/4a9b7bf9-795f-47ed-8a92-82356d4c5210/resourceGroups/mr-play-A_acsk8s_westus2/providers/Microsoft.Compute/virtualMachines/15216k8s9001' at line '1' and column '24520'. '**The template variable 'subnet' is not found**. Please see https://aka.ms/arm-template/#variables for usage details.'\"\r\n }\r\n}"}

**Comment by rtcn2** [comment]
Doesn't work for me , I run: 
 '```
az acs create --orchestrator-type kubernetes
 -g rn-jupyterhub-dev-rg-ne
 --master-vnet-subnet-id "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/dev-vnet-rg-ne/providers/Microsoft.Network/virtualNetworks/dev-vnet/subnets/k8s-master-dev-subnet"
 --agent-vnet-subnet-id "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/dev-vnet-rg-ne/providers/Microsoft.Network/virtualNetworks/dev-vnet/subnets/k8s-agent-dev-subnet"
 --master-first-consecutive-static-ip "10.12.24.5" --name rn-jupyterhub-dev-k8s'

but the vms are deployed to vnet 'k8s-vnet-xxxxxxxx' in the 'k8s-subnet', in North Europe
```

**Comment by Diros1234** [comment]
We need to create a Kubernetes cluster linked to a vnet where other services are running and we see that the CLI offers the options `master-vnet-subnet-id' & `agent-vnet-subnet-id`. However, those fields are ignores.
- When setting an existing vnet, this is just ignored a new vnet is created with network 10.0.0.0/8
- When setting a non-existing vnet, no error is shown, and again, a new vnet 10.0.0.0/ is created.

Here is the command we used:

az acs create --orchestrator-type kubernetes --name cluster-kubernetes-acs --resource-group pinakes-int-kubernetes-acs 
--master-vnet-subnet-id '/subscriptions/xxxxxxxxxxxx/resourceGroups/test/providers/Microsoft.Network/virtualNetworks/kubernetes-test-acs' 
--agent-vnet-subnet-id '/subscriptions/xxxxxxxxxxxxx/resourceGroups/test/providers/Microsoft.Network/virtualNetworks/kubernetes-test-acs' 
--master-first-consecutive-static-ip 10.3.0.1 --generate-ssh-keys


PS: network peering is not possible because the default K8s vnet is 10.0.0.0/8 which overlaps with other networks.

**Comment by hafizullah** [comment]
FYI this is only available in certain regions so if its not working that would be why.

**Comment by willwhite1** [comment]
When specifying vnet's I get the following error -

at line '1' and column '24298'. 'The template variable 'subnet' is not found

I looked in the generated ARM template from the ACS process and a variable subnet is referenced which doesn't exist. It looks like it should be feeding the master network CIDR into the CSE extension.

I am working in UKSOUTH which apparently should be supported. This looks like a bug in the way the ARM template is generated by ACS.
